src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,145 @@
+package duke;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The Parser class, containing the parsing logic for the bot.
+ *
+ * @author Jet Tan
+ */
+public class Parser {
+    /**
+     * Processes the input.
+     *
+     * @param input input of the user
+     * @throws DukeException duke.InvalidInputException, EmptyDescDescription, UnknownCommandException
+     */
+    public static void process(String input) throws DukeException, IOException {
+        String[] arr = input.split(" ");
+        String command = arr[0]; // first word of the user input
+        switch (command) {
+        case "bye":
+            Ui.exit();
+            break;
+        case "list":
+            if (TaskList.getTasks().size() == 0) {
+                System.out.println(Ui.getBorder() + "The list is empty. Why not add some tasks?\n" + Ui.getBorder());
+            } else {
+                StringBuilder listString = new StringBuilder();
+                for (int i = 0; i < TaskList.getTasks().size(); i++) {
+                    Task t = TaskList.getTasks().get(i);
+                    listString.append(i + 1).append(".").append(t.toString()).append("\n");
+                }
+                System.out.println(Ui.getBorder() + listString + Ui.getBorder());
+            }
+            break;
+        case "mark": {
+            int num = Integer.parseInt(arr[1]);
```

Comment:
just a suggestion maybe each case of the switch statements can be executed by a private function (each for a command) defined in parser class.

For example, for **case "mark":** , a function **_private void markTask(int num)_**, could handle the code execution. it may make it easier to read and also edit in the future.... what do u think ? 🤔